python/lsst/obs/base/fitsRawFormatterBase.py

```diff
@@ -118,21 +125,82 @@ def makeVisitInfo(self, metadata):
         """
         raise NotImplementedError("Must be implemented by subclasses.")
 
-    def makeWcs(self, metadata):
-        """Construct a SkyWcs from metadata.
+    @abstractmethod
+    def getDetector(self, id):
+        """Return the detector that acquired this raw exposure.
 
         Parameters
         ----------
+        id : `int`
+            The identifying number of the detector to get.
+
+        Returns
+        -------
+        detector : `~lsst.afw.cameraGeom.Detector`
+            The detector associated with that ``id``.
+        """
+        raise NotImplementedError("Must be implemented by subclasses.")
+
+    def makeWcs(self, visitInfo, detector, metadata):
+        """Construct a SkyWcs from the visitInfo and detector, or metadata
+        of an exposure.
+
+        If the boresight or camera geometry do not support this method of
+        WCS creation, this falls back on the header metadata-based version
+        (typically a purely linear FITS crval/crpix/cdmatrix WCS).
+
+        Parameters
+        ----------
+        visitInfo : `~lsst.afw.image.VisitInfo`
+            Description
+        detector : `~lsst.afw.cameraGeom.Detector`
+            Description
         metadata : `~lsst.daf.base.PropertyList`
             Header metadata.  May be modified in-place.
 
         Returns
         -------
-        wcs : `~lsst.afw.geom.SkyWcs`
+        skyWcs : `~lsst.afw.geom.SkyWcs`
             Reversible mapping from pixel coordinates to sky coordinates.
         """
-        from lsst.afw.geom import makeSkyWcs
-        return makeSkyWcs(metadata, strip=True)
+        skyWcs = self._createSkyWcsFromMetadata(metadata)
```

Comment:
Should this call happen later, so it can only fire if we fail to get the WCS from boresight and cameraGeom?

Reply:
I put this in first because that ensures that the metadata always gets stripped.